services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidateSimpleFlowSpec.groovy

```diff
@@ -0,0 +1,517 @@
+- excess - those meters/rules, which is exist on a switch, but it is NOT exist in db
+- proper - meters/rules value is the same on a switch and in db
+""")
+class MetersValidateSimpleFlowSpec extends BaseSpecification {
+    @Autowired
+    SwitchHelper switchHelper
+
+    def "Able to get meter validate information"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def defaultMetersSrcSwitch = northbound.getAllMeters(srcSwitch.dpId)
+        def defaultMetersDstSwitch = northbound.getAllMeters(dstSwitch.dpId)
+        def flow = flowHelper.addFlow(flowHelper.randomFlow(srcSwitch, dstSwitch))
+
+        then: "One meter is automatically created on the both switches"
+        def createdMetersIdSrcSwitch = getCreatedMetersId(srcSwitch.dpId, defaultMetersSrcSwitch)
+        def createdMetersIdDstSwitch = getCreatedMetersId(dstSwitch.dpId, defaultMetersDstSwitch)
+        def burstSizeSrcSwitch = switchHelper.getExpectedBurst(srcSwitch.dpId, flow.maximumBandwidth)
+        def burstSizeDstSwitch = switchHelper.getExpectedBurst(dstSwitch.dpId, flow.maximumBandwidth)
+
+        createdMetersIdSrcSwitch.size() == 1
+        createdMetersIdDstSwitch.size() == 1
+
+        and: "The correct info is stored in the validate meter response"
+        def meterValidateInfoSrcSwitch = northbound.validateMeters(srcSwitch.dpId)
+        def meterValidateInfoDstSwitch = northbound.validateMeters(dstSwitch.dpId)
+        def createdCookieSrcSwitch = getCreatedCookies(srcSwitch.dpId, createdMetersIdSrcSwitch)
+        def createdCookieDstSwitch = getCreatedCookies(dstSwitch.dpId, createdMetersIdDstSwitch)
+
+        def createdCookies = createdCookieSrcSwitch + createdCookieDstSwitch
+
+        meterValidateInfoSrcSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdSrcSwitch)
+        meterValidateInfoDstSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdDstSwitch)
+
+        meterValidateInfoSrcSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieSrcSwitch)
+        meterValidateInfoDstSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieDstSwitch)
+
+        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
+            it.meters.proper.each {
+                assert it.rate == flow.maximumBandwidth
+                assert it.flowId == flow.id
+            }
+        }
+
+        meterValidateInfoSrcSwitch.meters.proper.burstSize[0] == burstSizeSrcSwitch
+        meterValidateInfoDstSwitch.meters.proper.burstSize[0] == burstSizeDstSwitch
+
+        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
+            it.meters.proper.every {
+                ["KBPS", "BURST", "STATS"].containsAll(it.flags)
+            }
+        }
+
+        [srcSwitch, dstSwitch].each {
+            checkNoMeterValidation(it.dpId, ["proper"])
```

Comment:
method name is a non-sense for me. Not intuitive.

Reply:
You already have switch validation results. Why you call for it again inside the method? Just process the results that you already have, nothing changed since the last time you requested them

Reply:
checkNoMeterValidationExist?

Reply:
The method itself in current implementation cannot exist. It calls for validation multiple times for each 'section' which should not happen, just store the results and reuse them. So I can't really help you to rename it.
My implementation closest to what you have would be not to use any additional methods and just do this directly in test
```
["missing", "misconfigured", "excess"].each { section ->
   [srcSwitchValidateInfo, dstSwitchValidateInfo].each { validateResult ->
       assert validateResult.meters."$section".empty
  }
}
```

Reply:
fixed